action/protocol/context.go

```diff
@@ -147,3 +182,83 @@ func MustGetActionCtx(ctx context.Context) ActionCtx {
 	}
 	return ac
 }
+
+// WithFeatureCtx add FeatureCtx into context.
+func WithFeatureCtx(ctx context.Context) context.Context {
+	g := genesis.MustExtractGenesisContext(ctx)
+	height := MustGetBlockCtx(ctx).BlockHeight
+	return context.WithValue(
+		ctx,
+		featureContextKey{},
+		FeatureCtx{
+			DepositGasLast:            g.IsPacific(height),
+			SystemWideActionGasLimit:  !g.IsAleutian(height),
+			NotFixTopicCopyBug:        !g.IsAleutian(height),
+			SetRevertMessageToReceipt: g.IsHawaii(height),
+			FixGetHashFnHeight:        g.IsHawaii(height),
+			UsePendingNonceOption:     g.IsHawaii(height),
+			AsyncContractTrie:         g.IsGreenland(height),
+			StoreOutOfGasToReceipt:    !g.IsGreenland(height),
+			RefundAllDeposit:          g.IsPacific(height),
```

Comment:
`RefundAllDeposit` is the same fix as `DepositGasLast`
rename to `FixDoubleChargeGas`